modules/button/react/lib/types.ts

```diff
@@ -96,17 +96,22 @@ export interface ButtonColors {
 /**
  * Used to get the props of the anchor version of a button
  */
-export type AnchorButtonProps<P> = Omit<P, keyof React.ButtonHTMLAttributes<HTMLButtonElement>> &
+export type AnchorButtonProps<P, K extends keyof P = never> = Omit<
+  P,
+  keyof Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, K>
+> &
```

Comment:
I added `K` to allow for omitting specific keys from the `P` type. This prevents the need for defining this interface again